addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/CrossDomainScriptInclusionScanRule.java

```diff
@@ -51,16 +54,23 @@ public class CrossDomainScriptInclusionScanRule extends PluginPassiveScanner {
             LogManager.getLogger(CrossDomainScriptInclusionScanRule.class);
     private Model model = null;
 
+    private final TrustedDomains trustedDomains = new TrustedDomains();
+
     @Override
     public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+        trustedDomains.update(getConfig().getString(RuleConfigParam.RULE_DOMAINS_TRUSTED, ""));
+        trustedDomains.add(new RegexTrust(msg.getRequestHeader().getHostName()));
+
         if (msg.getResponseBody().length() > 0 && msg.getResponseHeader().isHtml()) {
             List<Element> sourceElements = source.getAllElements(HTMLElementName.SCRIPT);
             if (sourceElements != null) {
                 for (Element sourceElement : sourceElements) {
                     String src = sourceElement.getAttributeValue("src");
                     if (src != null
-                            && isScriptFromOtherDomain(
-                                    msg.getRequestHeader().getHostName(), src, msg)) {
+                                    && isScriptFromOtherDomain(
+                                            msg.getRequestHeader().getHostName(), src, msg)
+                            || !trustedDomains.isIncluded(
```

Comment:
This part of the code is returning null. What should be the input parameter for the function `.isIncluded`. I am using `.getAttributeValue("href")` which is returning 'null'.